src/main/java/com/amazonaws/samples/beam/taxi/count/TaxiCount.java

```diff
@@ -52,24 +57,33 @@ public static void main(String[] args) {
 
     Pipeline p = Pipeline.create(options);
 
-
     LOG.info("Running pipeline with options: {}", options.toString());
 
+    if (!StringUtils.isNullOrEmpty(options.getCodeGuruProfilingGroupName())) {
```

Comment:
This looks like it's only executed in the main method and hence only monitors the job manager. Can you please ensure that it's also executed on the job managers, eg, by adding a static method to the DoFns that initializes the profiler.